internal/pkg/admission/policy-server-deployment_test.go

```diff
@@ -210,38 +210,187 @@ func TestIfPolicyServerImageChanged(t *testing.T) {
 	}
 	if deployment.Spec.Template.Spec.Containers[0].SecurityContext.RunAsNonRoot == nil ||
-		*deployment.Spec.Template.Spec.Containers[0].SecurityContext.RunAsNonRoot == false {
-		t.Error("Policy server container should not run with root user")
+		*deployment.Spec.Template.Spec.Containers[0].SecurityContext.RunAsNonRoot != runAsRoot {
+		t.Error("Policy server container RunAsNonRoot diverges from the expected value")
 	}
 	if deployment.Spec.Template.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation == nil ||
-		*deployment.Spec.Template.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation == true {
-		t.Error("Policy server container should not run in container allowed to escalate privileges")
+		*deployment.Spec.Template.Spec.Containers[0].SecurityContext.AllowPrivilegeEscalation != allowPrivilegeEscalation {
+		t.Error("Policy server container AllowPrivilegeEscalation diverge from the expected value")
 	}
 	if deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities == nil ||
 		len(deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Add) > 0 ||
 		len(deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Drop) != 1 ||
 		deployment.Spec.Template.Spec.Containers[0].SecurityContext.Capabilities.Drop[0] != "all" {
-		t.Error("Policy server container should drop all capabilities")
+		t.Error("Policy server container Capabilities diverge from the expected value")
+	}
+}
+
+func TestPolicyServerPodSecurityContext(t *testing.T) {
+	reconciler := Reconciler{
+		Client:               nil,
+		DeploymentsNamespace: "kubewarden",
+	}
+	var user, group int64
+	user = 1000
+	group = 2000
+	podSecurity := corev1.PodSecurityContext{
+		RunAsUser:  &user,
+		RunAsGroup: &group,
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image: "image",
+			Security: policiesv1.PolicyServerSecurity{
+				Pod: &podSecurity,
+			},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+
+	if deployment.Spec.Template.Spec.SecurityContext == nil {
+		t.Error("Pod securityContext should be defined ")
+	}
+
+	if deployment.Spec.Template.Spec.SecurityContext != nil {
+		if *deployment.Spec.Template.Spec.SecurityContext.RunAsUser != user {
+			t.Error("Pod RunAsUser diverges from the expected value")
+		}
+		if *deployment.Spec.Template.Spec.SecurityContext.RunAsGroup != group {
+			t.Error("Pod RunAsGroup diverges from the expected value")
+		}
+	}
+}
+
+func TestPolicyServerWithNoSecurityContext(t *testing.T) {
+	reconciler := Reconciler{
+		Client:               nil,
+		DeploymentsNamespace: "kubewarden",
+	}
+	policyServer := &policiesv1.PolicyServer{
+		Spec: policiesv1.PolicyServerSpec{
+			Image:    "image",
+			Security: policiesv1.PolicyServerSecurity{},
+		},
+	}
+	deployment := reconciler.deployment("v1", policyServer)
+
+	if deployment.Spec.Template.Spec.SecurityContext != nil {
```

Comment:
Based on a previous comment, I think we should set a good default when the PolicyServer CRD doesn't provide guidance. This test should make sure our defaults are picked.

Reply:
Done